File: frontend/src/utils/teamMatch.ts
export function normalizeTeam(value: string) {
  return value
    .normalize("NFD")
    .replace(/[\u0300-\u036f]/g, "")
    .toLowerCase()
    .replace(/[^a-z0-9]/g, "");
}

export function sameTeam(left: string, right: string) {
  const normalizedLeft = normalizeTeam(left);
  const normalizedRight = normalizeTeam(right);
  return normalizedLeft === normalizedRight ||
    (Math.min(normalizedLeft.length, normalizedRight.length) >= 5 &&
      (normalizedLeft.includes(normalizedRight) || normalizedRight.includes(normalizedLeft)));
}

File: backend/app/utils/team_match.py
from __future__ import annotations

import re
import unicodedata


def normalize_team(value: str) -> str:
    normalized = unicodedata.normalize("NFKD", value.strip().lower())
    ascii_text = "".join(char for char in normalized if not unicodedata.combining(char))
    return re.sub(r"[^a-z0-9]", "", ascii_text)


def same_team(left: str, right: str) -> bool:
    normalized_left = normalize_team(left)
    normalized_right = normalize_team(right)
    if not normalized_left or not normalized_right:
        return False
    if normalized_left == normalized_right:
        return True
    shortest = min(len(normalized_left), len(normalized_right))
    return shortest >= 5 and (
        normalized_left in normalized_right or normalized_right in normalized_left
    )
